mm2src/coins/utxo.rs

```diff
@@ -483,6 +484,71 @@ impl UtxoSyncStatusLoopHandle {
     }
 }
 
+#[derive(Debug, Clone, Deserialize)]
+#[serde(tag = "type")]
+pub enum StartingHeader {
+    #[serde(rename = "genesis")]
+    Genesis { hex: String },
+    #[serde(rename = "other")]
+    Other { height: u64, hex: String },
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct SPVConf {
+    /// Where to start block headers sync from.
+    starting_header: StartingHeader,
+    /// Max number of block headers to be stored in db, when reached, old header will be deleted.
+    max_stored_block_headers: Option<NonZeroU64>,
+    /// The parameters that specify how the coin block headers should be validated. If None,
+    /// headers will be saved in DB without validation, can be none if the coin's RPC server is trusted.
+    validation_params: Option<BlockHeaderValidationParams>,
+}
+
+impl SPVConf {
+    pub fn starting_block_height(&self) -> u64 {
+        match self.starting_header {
+            StartingHeader::Genesis { .. } => 0,
+            StartingHeader::Other { height, .. } => height,
+        }
+    }
+
+    pub fn starting_block_header_hex(&self) -> String {
+        match self.starting_header.clone() {
+            StartingHeader::Genesis { hex } => hex,
+            StartingHeader::Other { hex, .. } => hex,
+        }
+    }
+
+    pub fn validate_starting_block_header(&self, coin: &str) -> Result<(), SPVError> {
+        // Check if verification params is set. If not set then we can return Ok response.
+        if self.validation_params.is_none() {
+            return Ok(());
+        }
+
+        let retarget_interval = RETARGETING_INTERVAL as u64;
```

Comment:
`RETARGETING_INTERVAL` is specific to `BTC`, this function should work for all coins. We should find another way to validate the starting block header based on the coin.

Reply:
Please consider the following changes https://github.com/KomodoPlatform/atomicDEX-API/compare/spv-optimizations...KomodoPlatform:atomicDEX-API:validate-starting-header-diff-algo
The PR still has a long way to go, I left 2 small `Todo` in the code, please also fix `test_block_header_utxo_loop` again